amundsen_application/api/utils/metadata_utils.py

```diff
@@ -42,7 +43,17 @@ def marshall_table_full(table_dict: Dict) -> Dict:
     table: Table = schema.load(table_dict).data
     results: Dict[str, Any] = schema.dump(table).data
 
-    is_editable = results['schema'] not in app.config['UNEDITABLE_SCHEMAS']
+    # Check if schecma is uneditable
+    is_editable_schema = results['schema'] not in app.config['UNEDITABLE_SCHEMAS']
+
+    # Check if Table is uneditable
+    table_id = results['schema'] + "." + results['name']
```

Comment:
I have concerns about applying the regex to the pattern of `schema.name`. This is a pattern that we have hardcoded in certain places because this is how our current design creates a display name for the tables. The concern is that this pattern would not be applicable for others in open source who aren't using the `schema.name` pattern.

However, the proposal to add `is_editable` to the resource metadata itself as our long term approach would address this concern since it would deprecate this logic. So I'm raising this not to request changes, but to acknowledge another benefit of moving to the long term approach as soon as we are able to.

Reply:
@ttannis @feng-tao , how should I proceed with this? One way I can think of eliminating this company specific `table_id` from front-end is by adding three variables in config files : 
1 .one variable is table_metadata attributes which we want to be part of our matching criteria (table_id) e.g `table_id_constituents = ['schema' , 'name', 'source', 'last_updated_timestamp']`
2. Another variable in config which will define operators to combine above defined table_id_constituents, e.g `table_id_operators = ['.', '_', '.']`
Length of `table_id_operators` array will be one less than `table_id_constituents` array. 
3. Third variable would be regex string with which want to match our `table_id`.
It is not good solution as backend one but let me know your thoughts if I should try something like this, or is it better to start from backend itself.

Reply:
I think to affectively address the use cases we might have to take a step back and do a new approach. I think we have to restructure the configuration so that users can have better control over what combinations of schemas and table_names they want to set as un-editable. The use cases are:
1. Set all tables with a given schema pattern as un-editable.
2. Set all tables with a specific table name  pattern in a given schema pattern as un-editable. This is the use case that creates the need for a new configuration structure, because users could want to apply different table patterns to different schemas.
3. Set all tables with a given table name pattern as editable.

In order to achieve the above via configurations, I'd propose switching to some kind of configuration where users would set a list of "match rules":
```
# MatchRuleObject could contain an optional regex for schemas and an option regex for table patterns

UNEDITABLE_TABLE_DESCRIPTION_MATCH_RULES: List[MatchRuleObject] = [
  # use case 1 - all tables in "schema" and "schema1" uneditable. 
  # An alternative to `UNEDITABLE_SCHEMAS`, but we could still keep that config for backwards compatibility 
  # until we switch to the long term solution
  { schema_regex: r”^(schema|schema1)" }, 
  
  # use case 2 - all "test_*" tables in "schema2" uneditable
  { schema_regex: r”^(schema2)", table_name_regex: r"^test_([a-zA-Z_0-9]+)" }, 
  
  # use case 2 - all "other_*" tables in "schema3" and "schema4" uneditable
  { schema_regex: r”^(schema3|schema4)", table_name_regex: r"^other_([a-zA-Z_0-9]+)" }, 
  
  # use case 3 - all "noedit_*" tables uneditable
  { table_name_regex: r"^noedit_([a-zA-Z_0-9]+)" }
]
```

What do you all think of this kind of configuration instead?

Reply:
This approach sounds good to me as far as we want to make decision only on basis of 'table name' and 'schema'. It is certainly better than fixing our pattern to 'schema.table_name', but it would not suffice if someone wants to make decision on the basis of other components of metadata like 'source'. We can move forward with this for the time till we move to long term solution.

Reply:
> but it would not suffice if someone wants to make decision on the basis of other components of metadata like 'source'

Yes. And that is okay for us because we acknowledged that this solution only needs to consider descriptions, beacause extended support to configure other editable structured metadata will deprecate these configurationsin favor of adding new attributes to the table metadata.